src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1156,6 +1159,47 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
     }
   }
 
+
+  //todo: clean this up, also accept a user and check if they can write to bucket, if they're in proxy, and if their proxy is in appropriate group
+  def getWorkspaceStatus(workspaceName: WorkspaceName, userSubjectId: RawlsUserSubjectId): Future[PerRequestMessage] = {
+    asAdmin {
+      dataSource.inFutureTransaction(readLocks = Set(workspaceName)) { txn =>
+        withWorkspaceContext(workspaceName, txn) { workspaceContext =>
+          val user = containerDAO.authDAO.loadUser(RawlsUserRef(userSubjectId), txn).get
+          val maxessLevel = containerDAO.authDAO.getMaximumAccessLevel(user, workspaceContext.workspace.workspaceId, txn)
+          val bucketName = workspaceContext.workspace.bucketName
+          val accessLevels = WorkspaceAccessLevels.groupAccessLevelsAscending
+
+          val rawlsGroupStatuses = accessLevels.map { lvl =>
+            s"RAWLS_${lvl}_EXISTS" -> !containerDAO.authDAO.loadGroup(RawlsGroupRef(RawlsGroupName(UserAuth.toWorkspaceAccessGroupName(workspaceName, lvl))), txn).isEmpty
+          }
+
+          val googleGroupStatuses = accessLevels.map { lvl =>
```

Comment:
this is going to change shortly, gcsDAO will no longer have toGroupId, getGoogleGroup should take RawlsGroup as a parameter. The access group refs are stored on the workspace, load them to get the actual groups.